core/src/main/scala/org/apache/spark/sql/delta/optimizer/OptimizeMetadataOnlyDeltaQuery.scala

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta.optimizer
+
+import org.apache.spark.sql.{Column, Row}
+import org.apache.spark.sql.catalyst.expressions.aggregate.Complete
+import org.apache.spark.sql.catalyst.expressions.{Alias, AttributeReference, Literal, SubqueryExpression}
+import org.apache.spark.sql.catalyst.expressions.aggregate.{AggregateExpression, Count}
+import org.apache.spark.sql.catalyst.planning.PhysicalOperation
+import org.apache.spark.sql.catalyst.plans.logical._
+import org.apache.spark.sql.delta.DeltaTable
+import org.apache.spark.sql.delta.files.TahoeLogFileIndex
+import org.apache.spark.sql.delta.stats.DeltaScanGenerator
+import org.apache.spark.sql.functions.{count, sum}
+import org.apache.spark.sql.types.LongType
+
+trait OptimizeMetadataOnlyDeltaQuery {
+  def optimizeQueryWithMetadata(plan: LogicalPlan): LogicalPlan = {
+    plan.transformUpWithSubqueries {
+      case CountStarDeltaTable(alias, countValue) =>

```

Comment:
I just tried to clean up the code a bit using the existing `Aggregate` like this:
```scala
  def optimizeQueryWithMetadata(plan: LogicalPlan): LogicalPlan = {
    plan.transformUpWithSubqueries {
      case agg @ CountStarDeltaTable(countValue) =>
        LocalRelation(agg.output, Seq(InternalRow(countValue)))
    }
  }

  protected def getDeltaScanGenerator(index: TahoeLogFileIndex): DeltaScanGenerator

  object CountStarDeltaTable {

    def unapply(plan: Aggregate): Option[Long] = {
      plan match {
        case Aggregate(
            Nil,
            Seq(Alias(AggregateExpression(Count(Seq(Literal(1, _))), Complete, false, None, _), _)),
            PhysicalOperation(_, Nil, DeltaTable(tahoeLogFileIndex: TahoeLogFileIndex))) =>
          extractGlobalCount(tahoeLogFileIndex)
        case _ => None
      }
    }
```

Could you update this and also clean up the unused imports in this file after that?

Reply:
Nice clean up. Thanks